corehq/apps/accounting/forms.py

```diff
@@ -566,7 +566,7 @@ def clean(self):
             and start_date > self.cleaned_data['end_date']):
             raise ValidationError("End date must be after start date.")
 
-        if self.cleaned_data['end_date'] <= datetime.date.today():
+        if self.cleaned_data['end_date'] and self.cleaned_data['end_date'] <= datetime.date.today():
```

Comment:
I recommend punting on most billing timezone issues, since if we're going to standardize them, first we need to answer what end date means for the user - right now subscriptions start and end at 00:00 UTC, regardless of what timezone the users of the project space have.  I'd say standardizing this is a low to low-medium priority compared to other pending billing work.